InterviewFlashCards/JsonFileReader.swift

```diff
@@ -22,14 +22,23 @@ struct JsonFileReader: DataGenerator {
         guard let path = fileName.resourcePath,
             let jsonData = try? Data(contentsOf: path,
                                      options: .mappedIfSafe),
-            let json = (try? JSONSerialization
-                .jsonObject(with: jsonData as Data,
-                            options: .allowFragments)) as? [String: Any],
-            let data = json["data"] as? JSON else {
+            let json = json(from: jsonData),
+            let data = json["data"] as? [JSON] else {
                 completion([])
```

Comment:
what's going on here? i thought an empty array was being passed into the completion but the array has no type so could you explain please?